packages/babel-types/src/definitions/core.js

```diff
@@ -310,7 +310,6 @@ defineType("IfStatement", {
       validate: assertNodeType("Expression")
     },
     consequent: {
-      optional: true,
```

Comment:
This is required but I guess this wouldn't ever cause an error since you can't do it -  `if (true) {}`